apostrophecms/apostrophe
# implement backlinks of related docs

[PRO-2575](https://linear.app/apostrophecms/issue/PRO-2575/as-a-developer-i-can-find-the-documents-that-have-relationships-to-a)

## Summary

When saving a doc, update its related documents' `relatedReverseIds` field accordingly.
For instance, when adding related topic(s) to an article, we add the article ID to the topic(s)' `relatedReverseIds` field.

This is done by the `updateBacklinks` beforeSave handler.

## What are the specific steps to test this change?

- Add some related docs to a doc's `relationship` field (for instance, add some topics to an article, if you follow the example from the [doc](https://v3.docs.apostrophecms.org/guide/relationships.html#the-anatomy-of-a-relationship-field))

<img width="990" alt="image" src="https://user-images.githubusercontent.com/8301962/159893626-9093e469-5feb-4860-a158-5f416a373ab6.png">

- Check that for all these related docs, their `relatedReverseIds` field now have the doc ID in the DB

<img width="321" alt="image" src="https://user-images.githubusercontent.com/8301962/159893763-7f59bd80-0876-421a-b2c6-04819ea07d93.png">

- Removing related docs should also remove the doc ID from their `relatedReverseIds` field :

<img width="993" alt="image" src="https://user-images.githubusercontent.com/8301962/159894339-815bd664-444a-49b6-9e06-6d26abe8527d.png">

<img width="345" alt="image" src="https://user-images.githubusercontent.com/8301962/159893977-a639af2d-ff6c-4d59-878d-9db1f8f68849.png">

## What kind of change does this PR introduce?
*(Check at least one)*


## Make sure the PR fulfills these requirements:


If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**
